### PR TITLE
🐛 linux: gate the su-restriction check on PAM config, simplify it, bump to 2.7.1

### PR DESCRIPTION
**Pull request by tas50**
Two changes to the **"access to the su command is restricted"** check in `mondoo-linux-security`, plus a patch version bump.

## 1. Gate on PAM config being present

```diff
     filters: |
       asset.kind != "container-image"
+      pam.conf.exists
```

The check queries `pam.conf.entries["/etc/pam.d/su"]`. On hosts that ship **no** PAM configuration — container-optimized k8s node images (COS, Flatcar, Bottlerocket) have no `/etc/pam.d` — that lookup yields a typed null, which is what surfaced the scan-crashing engine panic in the recent incident (hardened in mql #8320 / #8321). Gating on `pam.conf.exists` makes those hosts correctly **not-applicable** instead of failed/errored.

## 2. Simplify the group-existence assertion

```diff
       suRestrictedPam != empty
-      suRestrictedGroups == empty || groups.map(name).containsAll(suRestrictedGroups)
+      groups.map(name).containsAll(suRestrictedGroups)
       suRestrictedGroups != empty || groups.any(name.in(["wheel", "sudo"]))
```

The `suRestrictedGroups == empty ||` short-circuit is no longer needed:
- `containsAll([])` is vacuously true (the no-explicit-`group=` case), and
- `containsAll(null)` is now a safe `false` rather than a panic (mql #8320) — and on that path `suRestrictedPam != empty` already fails the check.

Overall pass/fail is unchanged. This is the only check in the bundle using the `pam` resource.

## 3. Version bump

`mondoo-linux-security` `2.7.0` → `2.7.1`.

## Dependency

Both `pam.conf.exists` (os provider, mql #8322) and the safe `containsAll(null)` behavior (mql #8320) ship in mql. This policy should land/release **after** the mql release that includes them; lint against an older provider fails with `cannot find field 'exists' in pam.conf`.

## Testing

- `cnspec policy lint content/mondoo-linux-security.mql.yaml` → valid (against an os provider build with the new field).
- Negative control: a bogus filter field fails lint with `cannot find field …`, confirming filters compile against the provider schema.

### CODE REVIEW

File: content/mondoo-linux-security.mql.yaml [hunk 1]
```diff
@@ -13920,12 +13920,13 @@ queries:
...
     filters: |
       asset.kind != "container-image"
+      pam.conf.exists
     mql: |
       suRestrictedPam = pam.conf.entries["/etc/pam.d/su"].where(pamType == "auth" && module == "pam_wheel.so" && params["use_uid"] != null)
       suRestrictedGroups = suRestrictedPam.where(params["group"] != null).map(params["group"])
 
       suRestrictedPam != empty
-      suRestrictedGroups == empty || groups.map(name).containsAll(suRestrictedGroups)
+      groups.map(name).containsAll(suRestrictedGroups)
```
[the remark this answers is not in the document]
Comment (tas50) [reply content/mondoo-linux-security.mql.yaml:13929]: Confirmed — `containsAll` of an empty (typed, non-null) list is guaranteed `true` in this engine, so no regression.

In the check, when no `group=` is configured, `suRestrictedGroups = suRestrictedPam.where(params["group"] != null).map(params["group"])` is an empty `[]string` (not null). `groups.map(name).containsAll(<empty []string>)` is vacuously satisfied → the compiled `== []` is `true`.

Verified against the engine: `["wheel","sudo"].containsAll(a.where(false))` → `true`. I also added a locking regression test in mql (#8320) pinning exactly this (`core: lock containsAll-of-empty-list returns true`).

Note the empty-list case is distinct from a *null* arg (a `map[string][]T` key miss), which propagates null → `== []` is `false`; that path only occurs here when there's no `/etc/pam.d/su` at all, in which case `suRestrictedPam != empty` already fails the check. Full case matrix:
- su file + wheel line, no group= → suRestrictedGroups `[]` → line 2 true ✓
- su file + wheel line + group=X → checks X exists ✓
- su file, no wheel line → line 2 true, but `suRestrictedPam != empty` fails → check fails ✓
- no su file (null) → line 2 false (safe, no panic via #8320), and `suRestrictedPam != empty` also fails → check fails ✓

**Review by jaym — APPROVED** [review]

File: content/mondoo-linux-security.mql.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tas50) [reply content/mondoo-linux-security.mql.yaml:13929]: Confirmed: `containsAll([])` is `true` in MQL. When no `group=` is configured, `suRestrictedGroups` is an empty (typed, non-null) `[]string`, and `groups.map(name).containsAll(<empty>)` is vacuously satisfied → compiled `== []` is `true`. Verified against the engine (`["wheel","sudo"].containsAll(a.where(false))` → `true`) and pinned with a regression test in mql #8320 (`core: lock containsAll-of-empty-list returns true`). The only case that yields a null arg is a missing `/etc/pam.d/su` entirely, where `suRestrictedPam != empty` already fails the check — so no false failures.

**Comment by jaym** [comment]
Heads-up: v5.26.1 stopped the k8s-node panics (PAM-less hosts are excluded by the `pam.conf.exists` filter — fleet rate dropped 164→27/15m at 20:15Z), but hosts that **have `/etc/pam.d/` and lack only the `su` file still panic**: the filter passes, `suRestrictedGroups` is null, and with the `== empty ||` disjunct removed the null now flows into `containsAll()` unconditionally.

Production: `sva-sap-server` (SLES SAP fleet, zero k8s assets) is still panicking post-rollout — events at 20:54:48Z/20:56:51Z on cnspec 13.22.0 show `runBoundFunction(…, 0x10000000e)`, which is the `containsAll` chunk position of the **v5.26.1-compiled** text (v5.26.0 compiled it at `0x100000010`). Reproduced with released 13.22.0 on ubuntu:24.04 and opensuse/leap:15.6 by removing `/etc/pam.d/su` while keeping `/etc/pam.d/` populated.

Follow-up fix: #2771 (if-guard, verified panic-free on the same matrix). Incident: INC-2026-06-10-client-panic-k8s.
